### ISSUE TITLE
value-keyword-case

### ISSUE THREAD

**Issue by FollowingFawn** [stylelint/stylelint#4647]
> Clearly describe the bug

e.g. "There are false positives for two leading zeros when ..."

> Which rule, if any, is the bug related to?

e.g. `number-leading-zero`

> What code is needed to reproduce the bug?

e.g.

```css
a {
  width: 00.1em;
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "number-leading-zero": "always"
  }
}
```

> Which version of stylelint are you using?

e.g. `9.2.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

e.g. "CLI with `stylelint "**/*.css" --config myconfig.json`"

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

e.g. "Yes, it's related to SCSS nested properties."

> What did you expect to happen?

e.g. "No warnings to be flagged."

> What actually happened (e.g. what warnings or errors did you get)?

e.g. "The following warnings were flagged:"

```shell
test.css
2:4    Expected a leading zero (number-leading-zero)
```
